app/controllers/concerns/potential_spam_user.rb

```diff
@@ -0,0 +1,21 @@
+# This module is the source of truth for what we consider a potential spam user.
+module PotentialSpamUser
+  extend ActiveSupport::Concern
+
+  # I think these should exist in constants even though they are evaluated once
+  # it makes it easier to tweak these variables consistently.
+  # rubocop:disable Rails/RelativeDateConstant
+  AUTH_ACCOUNT_AGE = 50.hours.ago
+  ACCOUNT_NAME_CHARACTER_LENGTH = 30
+  ACCOUNT_AGE = 48.hours.ago
+  LIMIT = 150
+  # rubocop:enable Rails/RelativeDateConstant
+
+  def potential_spam_users
+    User.where("github_created_at > ? OR twitter_created_at > ? OR length(name) > ?", ACCOUNT_AGE, ACCOUNT_AGE, ACCOUNT_NAME_CHARACTER_LENGTH).
+      where("created_at > ?", ACCOUNT_AGE).
+      order("created_at DESC").
+      where.not("username LIKE ?", "%spam_%").
```

Comment:
If prefixing usernames with `spam_` is something we do (I think it is), then we can probably anchor the query on the left by removing the initial `%`, which should actually use the index.

Reply:
Cool! I just pulled this out of the controller as it was, but that's a good improvement, I'll do that if I move this PR forward after talking to Peter.

Reply:
For reference:

> In addition, the query planner can use a B-tree index for queries that involve a pattern matching operator LIKE and ~ if the pattern is a constant and is anchor at the beginning of the pattern, for example:
>
> column_name LIKE 'foo%' 
> column_name LKE 'bar%' 
> column_name  ~ '^foo'

Note that depending on our locale the index might still not get picked up depending on the DB's locale. The relevant setting can be seen with `SHOW LC_COLLATE` in `psql` prompt.